indico/modules/rb_new/client/js/common/timeline/DailyTimelineContent.jsx

```diff
@@ -108,12 +108,19 @@ export default class DailyTimelineContent extends React.Component {
         );
     };
 
+    hasOccurrenceActions = (booking) => {
+        return Object.values(booking.occurrences.bookings).some(([occ]) => {
+            return (occ && (occ.canCancel || occ.canReject));
+        });
+    };
+
     get hasActions() {
-        const {rowActions} = this.props;
-        return Object.values(rowActions).includes(true);
+        const {booking, rowActions} = this.props;
+        const hasOccurrenceActions = rowActions.occurrence ? this.hasOccurrenceActions(booking) : false;
+        return rowActions.roomTimeline || hasOccurrenceActions;
     }
 
-    renderTimelineRow({availability, label, verboseLabel, room}, key, rowStyle = null) {
+    renderTimelineRow({availability, label, verboseLabel, room}, key, hasActions, rowStyle = null) {
```

Comment:
Can't you just use `this.hasActions` instead of passing it as an argument to these functions?

Reply:
I could, but in that case it would check the occurrences for every row, I figured it's better to check once